core/src/main/java/org/apache/brooklyn/util/core/internal/ssh/SshTool.java

```diff
@@ -49,7 +50,14 @@
      * These keys are detected from entity/global config and automatically applied to ssh executions. */
     public static final String BROOKLYN_CONFIG_KEY_PREFIX = "brooklyn.ssh.config.";
     
-    public static final ConfigKey<String> PROP_TOOL_CLASS = newStringConfigKey("tool.class", "SshTool implementation to use", null);
+    /**
+     * @deprecated since 0.9.0; use {@link SshMachineLocation#SSH_TOOL_CLASS}
+     */
+    @Deprecated
+    public static final ConfigKey<String> PROP_TOOL_CLASS = newStringConfigKey(
```

Comment:
Can you explain why did you deprecate this?
It is good if this key is in a generic location like `SshTool`. I was thinking about refactoring Winrm support to follow the same pattern as it is for ssh, if so we could implement this dynamic sshTool loading with the same keys and the same API.
In regard with this please check https://github.com/apache/incubator-brooklyn/pull/950